oxo.py

```diff
@@ -1,26 +1,81 @@
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return self.tile[x][y]
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        if self.tile[x][y] == self.empty:
+            self.tile[x][y] = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        for x in xrange(self.tile_width):
+            for y in xrange(self.tile_height):
+
+                if self.tile[x][y] == self.empty:
+                    return False
+
+        return True
 
     def get_winner(self):
         """ If a player has three in a row, return 1 or 2 depending on which player.
             Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+        for x in xrange(self.tile_width):
+            for y in xrange(self.tile_height):
+
+                # Checks if there are still empty tiles
+                # If there are - continue the loop
+                if self.tile[x][y] == self.empty:
+                    continue
+
+                # Loops through 4 different direction to determine
+                # whether a player has reached 3 in a row (winning condition)
+                for vector in [(1, 0), (1, 1), (0, 1), (-1, 1)]:
+                    try:
+                        tile_to_check = [x, y]
+                        mark_to_check_for = self.tile[x][y]
+
+                        for i in range(1, self.winning_condition):
+                            tile_to_check[0] += vector[0]
+                            tile_to_check[1] += vector[1]
+
+                            # Checks tiles to prevent unfair winning
+                            if self.tile[tile_to_check[0]][tile_to_check[1]]\
+                                    != mark_to_check_for:
+                                break
+
+                            # Checks tiles to prevent unfair winning
+                            if tile_to_check[0] >= self.tile_width or\
+                                    tile_to_check[1] >= self.tile_height or\
+                                    tile_to_check[0] < 0 or\
+                                    tile_to_check[1] < 0:
+                                break
+
+                            # Returns number of a winner (1 or 2)
+                            if i == (self.winning_condition - 1):
+                                return self.tile[x][y]
+
+                    except:
```

Comment:
It's generally bad practice to catch all exceptions -- you should specify which type of exception(s) you are catching, e.g. `except IndexError:`. But I'm not actually sure what you _are_ trying to catch here, since you already have array bounds checks etc?